src/nccl_ofi_rdma.cpp

```diff
@@ -6852,119 +6833,77 @@ nccl_net_ofi_rdma_ep_t::nccl_net_ofi_rdma_ep_t(nccl_net_ofi_rdma_domain_t *domai
 }
 
 
-static int
-nccl_net_ofi_rdma_domain_free(nccl_net_ofi_domain_t *base_domain)
+int nccl_net_ofi_rdma_domain_t::cleanup_resources()
 {
-	int ret;
-	nccl_net_ofi_rdma_domain_t *domain = (nccl_net_ofi_rdma_domain_t *)base_domain;
+	int ret = 0;
+	int err_code = 0;
 
-	if (domain->cm) {
-		delete domain->cm;
-		domain->cm = nullptr;
+	called_cleanup_resources = true;
+
+	if (this->cm) {
```

Comment:
Is there a reason we're still dynamically creating a CM here?  Obviously, the scheduler needs some work first, but feels like we don't need to be dynamic here.

Reply:
yeah, a todo is fine.

Reply:
Added TODO in latest force push. I also tried to move the CM from being members in both the SENDRECV and RDMA transport derived domain classes to just being in the base domain, but trying to include `#include "cm/nccl_ofi_cm.h"` in `nccl_ofi.h` failed due to the connection establishment code being dependent on macros and types defined in `nccl_ofi.h`. I'll make a note to see if I can move the CM to the base domain class further along in the refactoring process.